cirq/circuits/circuit.py

```diff
@@ -2001,12 +2000,17 @@ def with_noise(self, noise: 'cirq.NOISE_MODEL_LIKE') -> 'cirq.Circuit':
 
 
 def _resolve_operations(operations: Iterable['cirq.Operation'],
-                        param_resolver: 'cirq.ParamResolver'
-                       ) -> List['cirq.Operation']:
+                        param_resolver: 'cirq.ParamResolver',
+                        recursive: bool) -> List['cirq.Operation']:
     resolved_operations = []  # type: List['cirq.Operation']
-    for op in operations:
-        resolved_operations.append(protocols.resolve_parameters(
-            op, param_resolver))
+    if recursive:
+        for op in operations:
+            resolved_operations.append(
+                protocols.resolve_parameters(op, param_resolver))
+    else:
+        for op in operations:
+            resolved_operations.append(
+                protocols.resolve_parameters_once(op, param_resolver))
```

Comment:
This conditional on `recursive` flag is present throughout this PR which suggests that client code would be simpler if we combined `resolve_parameters` and `resolve_parameters_once` protocols into one that takes the recursive flag.

Reply:
Good point. I think I'll keep `resolve_parameters_once` as an alias for `resolve_parameters({...}, recursive=False)` though, since it has a nice parallel with `decompose[_once]`.

Reply:
This is now done.